### DISCUSSION TITLE
New user Q: "Could not locate Gemfile" on `bundle install`

### DISCUSSION THREAD

**Discussion by CharlesWiltgen** [tabler/tabler#1185] [Q&A] [answered]
Context: New to Tabler, have built non-trivial Node and PHP apps.

When I follow the installation instructions and run `bundle install` from the Tabler directory, I get the error `Could not locate Gemfile`.

I'm on an Apple Silicon Mac. I'm new to Ruby (didn't realize it was a dependency when I purchased), but I've verified that Ruby is installed (ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [arm64-darwin21]) and that bundle is installed (Bundler version 2.3.11 (2022-04-07 commit 9b6c5a801a)).

Any thoughts as to what I'm doing wrong would be helpful!

**Comment by dheineman** [comment, marked as answer]
Hey @CharlesWiltgen 

The installation instructions you are talking about (that use ruby/bundler) are required when you want to help develop tabler. I suspect the error might be related to the ruby version. You seem to be using 3.1 and as the readme states: `the recommended version is 2.7.6.`

You can try to install version 2.5 for example with homebrew by running `brew install ruby @2.5` or use a Ruby version management tool like [rbenv](https://github.com/rbenv/rbenv), [rvm](https://github.com/rvm/rvm) or [ruby-install](https://github.com/postmodern/ruby-install) to quickly be able to switch ruby version.

I use an Apple M1 and have  `ruby 2.7.4p191` installed using ruby-install myself.

However if you just want to install tabler for use in your project then please refer to the https://github.com/tabler/tabler#installation section in the readme. You can install tabler using npm, downloading the dist files or using the provided CDN links.

**Reply by CharlesWiltgen** [reply]
@dheineman, I really appreciate the guidance! The brew method failed (`Error: ruby@2.5 has been disabled because it is not supported upstream!`) and rbenv did too (2.5 end-of-life warning then `BUILD FAILED (macOS 12.5 using ruby-build 20220726)`). But with your insight that I can just use pre-built versions, I'll happily do that for now. Thanks again! ✌️

**Comment by ghost** [comment]
UNSCRIBE

**Comment by ghost** [comment]
UNSCRIBE 



From: "Charles Wiltgen" ***@***.***> 
To: "tabler" ***@***.***> 
Cc: "Subscribed" ***@***.***> 
Sent: Thursday, August 4, 2022 9:08:59 AM 
Subject: [tabler/tabler] New user Q: "Could not locate Gemfile" on `bundle install` (Discussion #1185) 





Context: New to Tabler, have built non-trivial Node and PHP apps. 

When I follow the installation instructions and run bundle install from the Tabler directory, I get the error Could not locate Gemfile . 

I'm on an Apple Silicon Mac. I'm new to Ruby (didn't realize it was a dependency when I purchased), but I've verified that Ruby is installed (ruby 3.1.2p20 (2022-04-12 revision 4491bb740a) [arm64-darwin21]) and that bundle is installed (Bundler version 2.3.11 (2022-04-07 commit 9b6c5a801a)). 

Any thoughts as to what I'm doing wrong would be helpful!
